src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -181,23 +213,28 @@ public InventorySyncStatistics getStatistics() {
      *     <li>Instantiate {@code supplyChannelKeyToId} map</li>
      *     <li>Create missing supply channels if needed</li>
      * </ul>
-     * Method returns {@link CompletionStage} of {@link Map} that contains mapping of supply channels' keys to ids.
-     * It may contain {@link SyncProblemException} when fetching supply channels results in exception.
+     * Method returns {@link CompletionStage} of {@link Optional} that may contain mapping of supply channels'
+     * keys to ids. An empty optional indicates that fetching existing channels failed.
      *
      * @param drafts {@link List} containing {@link InventoryEntryDraft} objects where missing supply channels can occur
-     * @return {@link CompletionStage} of {@link Map} that contain mapping of supply channels' keys to ids, or
-     *      {@link SyncProblemException} when fetching supply channels results in exception
+     * @return {@link CompletionStage} of {@link Optional} that may contain mapping of supply channels' keys to ids.
      */
-    private CompletionStage<Map<String, String>> populateSupplyChannels(@Nonnull final List<InventoryEntryDraft>
-                                                                            drafts) {
+    private CompletionStage<Optional<Map<String, String>>> findSupplyChannels(@Nonnull final List<InventoryEntryDraft>
```

Comment:
since it actually creates supply channels I would not use name which implies it only reads some entities